pkg/process/config/config.go

```diff
@@ -293,12 +293,26 @@ func NewAgentConfig(loggerName config.LoggerName, yamlPath, netYamlPath string)
 	}
 
 	if cfg.HostName == "" {
-		if ecsutil.IsFargateInstance() {
-			// Fargate tasks should have no concept of host names, so we're using the task ARN.
-			if taskMeta, err := ecsmeta.V2().GetTask(); err == nil {
-				cfg.HostName = fmt.Sprintf("fargate_task:%s", taskMeta.TaskARN)
-			} else {
-				log.Errorf("Failed to retrieve Fargate task metadata: %s", err)
+		if fargate.IsFargateInstance() {
+			// Fargate should have no concept of host names
+			// we set cfg.HostName depending on the orchestrator
+			switch fargate.GetOrchestrator() {
+			case fargate.ECS:
+				// Use the task ARN as hostname
+				if taskMeta, err := ecsmeta.V2().GetTask(); err == nil {
+					cfg.HostName = fmt.Sprintf("fargate_task:%s", taskMeta.TaskARN)
+				} else {
+					log.Errorf("Failed to retrieve ECS Fargate task metadata: %s", err)
+				}
+			case fargate.EKS:
+				// Use the node name as hostname
+				if nodename, err := fargate.GetEKSFargateNodename(); err == nil {
+					cfg.HostName = nodename
```

Comment:
Was this changed? I thought we were observing a hostname of `-`?

Reply:
Can we extract this whole block into it's own function, and add tests if possible 🙇

Reply:
no this wasn't changed, maybe the backend sends `-` when it cannot resolve the hostname ? will add a test